frontend/public/components/utils/storage-class-dropdown.tsx

```diff
@@ -8,6 +8,40 @@ import { isDefaultClass } from '../storage-class';
 
 /* Component StorageClassDropdown - creates a dropdown list of storage classes */
 
+const getUnorderedItems = (items, filter) =>
+  filter ?
+    items.reduce( (acc, resource) => ({
+      ...acc,
+      ...(filter(resource) && {
+        [resource.metadata.name]: {
```

Comment:
There is a ton of similarity between this object and the one below.  Can you share code via function call rather than copy & paste?